### ISSUE TITLE
Update components to work with react-intl v2-v5

### ISSUE THREAD

**Issue by jmsv6d** [cerner/terra-core#3208]
# Feature Request
In an endeavor to minimize the number of major version bumps we want to make the update for our components to work with react-intl v5 wherever possible.

## Description
For this update the peer dependencies need to be updated. `intlShape` Is no longer exported since v3, so the imports for that must be removed and PropType for intl object updated. An undocumented non-passive change was made in react-intl for the `FormattedMessage` component, when a function is passed as a child of the component the function is now given the param of an array of strings instead of the joined string. This causes many of the components that we are rendering in these functions to have prop type warnings as they are expecting a string. There are a few approaches that we could take to make these components continue to work as expected for all the react-intl version. I believe for most of these cases we can switch from using the `FormattedMessage` component to `injectIntl` + `intl.formatMessage()` or the function provided to `FormattedMessage` can do a check if the param is an array and join the strings, if so.

## Additional Context / Screenshots
I have begun this work at https://github.com/cerner/terra-core/tree/react-intl-upgrade
This work should be able to be completed and released prior to the framework portions of the upgrade being completed.
Note: terra-base and terra-i18n should not be updated with this change.
